dask/array/core.py

```diff
@@ -949,6 +949,17 @@ class Array(Base):
     _default_get = staticmethod(threaded.get)
     _finalize = staticmethod(finalize)
 
+    def __new__(cls, *args, **kwargs):
+        obj = super(Array, cls).__new__(cls)
+        cls.__init__(obj, *args, **kwargs)
```

Comment:
I think Python calls `__init__` automatically after `__new__`:

    If __new__() returns an instance of cls, then the new instance’s __init__()
    method will be invoked like __init__(self[, ...]), where self is the new instance
    and the remaining arguments are the same as were passed to __new__().

https://docs.python.org/2/reference/datamodel.html#object.__new__

So this will result in `__new__` getting called twice. I don't know if it's possible to achieve what you want (calling a plugin after `__init__`) with a hook here.